public/chat_header_button.tsx

```diff
@@ -54,6 +56,9 @@ export const HeaderChatButton = (props: HeaderChatButtonProps) => {
   const flyoutFullScreen = sidecarDockedMode === SIDECAR_DOCKED_MODE.TAKEOVER;
   const flyoutMountPoint = useRef(null);
 
+  const resetChatRef = useRef(props.assistantActions.resetChat);
+  resetChatRef.current = props.assistantActions.resetChat;
```

Comment:
Can we just use the `props.assistantActions.resetChat`?

Reply:
I've tested in my local machine, the `props.assistantActions.resetChat` will be changed after chat button render. The `getDataSourceId$` will be executed multi times. It's OK to change to use `props.assistantActions.resetChat` directly.